packages/clay-date-picker/src/ClayDatePicker.js

```diff
@@ -0,0 +1,384 @@
+			if (!useNative) {
+				setWeeks(currentMonth, firstDayOfWeek);
+			}
+		},
+		[currentMonth]
+	);
+
+	const classNames = classnames('input-group', {
+		[id]: id,
+	});
+
+	return (
+		<div className={classNames} ref={elementRef}>
+			<div className="input-group-item">
+				<InputDate
+					ariaLabel={ariaLabel}
+					currentTime={currentTime}
+					dateFormat={dateFormat}
+					defaultValue={defaultValue}
+					inputName={inputName}
+					onChange={inputChange}
+					placeholder={placeholder}
+					time={time}
+					timeFormat={timeFormat}
+					useNative={useNative}
+					value={inputValue}
+				/>
+				{!useNative && (
+					<div className="input-group-inset-item input-group-inset-item-after">
+						<Button
+							className="d-none d-md-inline-block"
+							icon="calendar"
+							onClick={handleCalendarButtonClicked}
+							spritemap={spritemap}
+							style="unstyled"
+						/>
+					</div>
+				)}
+			</div>
+			{!useNative && (
+				<Dropdown
+					containerRef={elementRef}
+					expanded={expanded}
+					onDocumentClick={() => setExpanded(false)}
+				>
+					<React.Fragment>
+						<div className="datepicker">
+							<DateNavigation
+								currentMonth={currentMonth}
+								months={months}
+								onDotClicked={() => {
+									setCurrentMonth(initialMonth);
+									setDaySelected(initialMonth);
+									setInputValue(initialMonth);
+								}}
+								onMonthChange={setCurrentMonth}
+								spritemap={spritemap}
+								years={years}
+							/>
+							<Weekdays
+								firstDayOfWeek={firstDayOfWeek}
+								weekdaysShort={weekdaysShort}
+							>
+								{({weekday, key}) => (
+									<Week key={key} weekday={weekday} />
+								)}
+							</Weekdays>
+							<div className="datepicker-body">
+								<Weekday weeks={weeks}>
+									{({day, key}) => (
+										<Day
+											day={day}
+											daySelected={daySelected}
+											key={key}
+											onClick={handleDayClicked}
+										/>
+									)}
+								</Weekday>
+							</div>
+						</div>
+						{(footerElement || time) && (
+							<div className="datepicker-footer">
+								{time && (
+									<TimePicker
+										currentTime={currentTime}
+										onTimeChange={setCurrentTime}
+										spritemap={spritemap}
+										timeFormat={timeFormat}
+									/>
+								)}
+								{!time &&
+									React.Children.only(
+										footerElement({spritemap})
+									)}
+							</div>
+						)}
+					</React.Fragment>
+				</Dropdown>
+			)}
+		</div>
+	);
+}
+
+ClayDatePicker.propTypes = {
+	/**
+	 * Aria label attribute for the button element.
+	 * @default undefined
+	 */
+	ariaLabel: PropTypes.string,
+
+	/**
+	 * Set the format of how the date will appear in the input element.
+	 * See available: https://momentjs.com/docs/#/parsing/string-format/
+	 * @default YYYY-MM-DD
+	 */
+	dateFormat: PropTypes.string,
+
+	/**
+	 * Set the initial value of the input.
+	 * @default undefined
+	 */
+	defaultValue: PropTypes.oneOfType([
```

Comment:
Just want to double check with you, but does this component function as an "uncontrolled" component? I want to make sure we using similar wording as react's API. They distinguish value vs defaultValue based on the nature of it being "controlled".

Reference: https://reactjs.org/docs/uncontrolled-components.html#default-values

Reply:
In fact, I did not know, from that moment on, that subsequent updates were out of control, thanks for the link. But the reason for this API escapes the idea of ​​being uncontrolled just to set an initial value, the rest is controlled by the component and we issue the `onValueChange`. I think this will cause confusion for React components... I'm going to remove that.

I will follow the same line as you are doing in the color picker.